### PR TITLE
NCF keras: Add custom loss and metrics

### PR DESCRIPTION
**Pull request by guptapriya**
Added custom loss (using model.add_loss) and custom metrics (using custom layer) so that we can account for weights in those computations.
The results with this change seem similar to those with custom training loop:

```
Epoch 1/10
994/994 [==============================] - 46s 47ms/step - loss: 0.2469 - HR_METRIC: 0.0104 - val_loss: 0.0000e+00 - val_HR_METRIC: 0.5334
Epoch 2/10
994/994 [==============================] - 29s 29ms/step - loss: 0.1873 - HR_METRIC: 0.0100 - val_loss: 0.0000e+00 - val_HR_METRIC: 0.5875
Epoch 3/10
994/994 [==============================] - 29s 29ms/step - loss: 0.1689 - HR_METRIC: 0.0098 - val_loss: 0.0000e+00 - val_HR_METRIC: 0.6166
Epoch 4/10
994/994 [==============================] - 29s 29ms/step - loss: 0.1569 - HR_METRIC: 0.0101 - val_loss: 0.0000e+00 - val_HR_METRIC: 0.6203
Epoch 5/10
994/994 [==============================] - 27s 27ms/step - loss: 0.1487 - HR_METRIC: 0.0100 - val_loss: 0.0000e+00 - val_HR_METRIC: 0.6273
Epoch 6/10
994/994 [==============================] - 27s 27ms/step - loss: 0.1426 - HR_METRIC: 0.0103 - val_loss: 0.0000e+00 - val_HR_METRIC: 0.6305
Epoch 7/10
994/994 [==============================] - 26s 27ms/step - loss: 0.1383 - HR_METRIC: 0.0101 - val_loss: 0.0000e+00 - val_HR_METRIC: 0.6353
Epoch 8/10
994/994 [==============================] - 26s 27ms/step - loss: 0.1349 - HR_METRIC: 0.0106 - val_loss: 0.0000e+00 - val_HR_METRIC: 0.6347
Epoch 9/10
994/994 [==============================] - 27s 27ms/step - loss: 0.1322 - HR_METRIC: 0.0099 - val_loss: 0.0000e+00 - val_HR_METRIC: 0.6366
Epoch 10/10
994/994 [==============================] - 26s 27ms/step - loss: 0.1297 - HR_METRIC: 0.0098 - val_loss: 0.0000e+00 - val_HR_METRIC: 0.6361
```

Also removed the cloning flag as cloning is False by default now.

### CODE REVIEW

File: official/recommendation/ncf_keras_main.py [hunk 1]
```diff
@@ -45,50 +45,32 @@
...
+  def __init__(self, params):
+    super(MetricLayer, self).__init__()
+    self.params = params
 
-    in_top_k = tf.cond(
-        is_training,
-        lambda: tf.zeros(shape=in_top_k.shape, dtype=in_top_k.dtype),
-        lambda: in_top_k)
+  def build(self, input_shape):
+    self.metric = tf.keras.metrics.Mean(name=rconst.HR_METRIC_NAME)
```
Comment (pavithrasv) [inline official/recommendation/ncf_keras_main.py]: We can create this in __init__ as this does not rely on input shape.

File: official/recommendation/ncf_keras_main.py [hunk 2]
```diff
@@ -45,50 +45,32 @@
...
 
-    return in_top_k
-
-  return metric_fn
+  def call(self, inputs):
+    logits, dup_mask = inputs
+    in_top_k, metric_weights = metric_fn(logits, dup_mask, self.params)
+    self.add_metric(self.metric(in_top_k, metric_weights))
+    return logits
```
Comment (pavithrasv) [inline official/recommendation/ncf_keras_main.py]: Can we pass this as sample_weight=metric_weights for kwargs for readability?

**Comment by pavithrasv** [comment]
Thank you for the PR!

**Comment by guptapriya** [comment]
Fixed the lint issue but cannot figure out why the other builds failed..
